### ISSUE TITLE
Batch logit processors crash with index out of range in concurrent mode

### ISSUE THREAD

**Issue by scouzi1966** [scouzi1966/maclocal-api#72]
## Bug

Logit processors (`TopKProcessor`, `PresenceContext`) crash with **Swift runtime failure: Index out of range** when used with `--concurrent N` (batch mode). The crash occurs in `BatchScheduler.prefillOne()` during the logit processing step.

## Reproduction

```bash
afm mlx -m mlx-community/Qwen3.5-35B-A3B-4bit -w --concurrent 10
# Send a request from WebUI with top_k or presence_penalty set (even to 0)
```

## Crash Stack

```
0: Swift runtime failure: Index out of range
1: _ArrayBuffer._checkValidSubscriptMutating(_:)
3: Array.subscript.modify
4: getItemND(src:operations:stream:)
5: MLXArray.subscript.getter
6: TopKProcessor.process(logits:)    ← or PresenceContext.process(logits:)
7: BatchScheduler.prefillOne(_:)
8: BatchScheduler.generationLoop()
```

## Root Cause

The logit processors in `Scripts/patches/Evaluate.swift` (`TopKProcessor`, `MinPProcessor`, `PresenceContext`, `CompositeLogitProcessor`) were written for single-sequence inference. In batch mode, `BatchScheduler.prefillOne()` passes logits with a batch dimension that these processors don't expect, causing array index out of bounds in MLX's `gatherND`/`getItemND`.

## Affected Processors

- `TopKProcessor.process(logits:)` — confirmed crash
- `PresenceContext.process(logits:)` — confirmed crash
- `MinPProcessor.process(logits:)` — likely affected (same pattern)

## Workaround

Don't send `top_k`, `min_p`, or `presence_penalty` parameters when using `--concurrent` mode. The WebUI sends these by default — set them to empty/disabled.

## Fix

The logit processors need to handle batched logits (2D tensor `[batch, vocab]`) in addition to single-sequence logits (`[1, vocab]` or `[vocab]`). Either:
1. Process each sequence in the batch independently
2. Or vectorize the operations across the batch dimension

Found during nightly v0.9.9-74dbbe2 testing.
